src/double64.c

```diff
@@ -282,148 +282,69 @@ double64_init	(SF_PRIVATE *psf)
 
 double
 double64_be_read (const unsigned char *cptr)
-{	int		exponent, negative, upper, lower ;
+{
 	double	dvalue ;
+	uint8_t *dbytes = (uint8_t *) &dvalue ;
```

Comment:
don't use `uint8_t`, which is not guaranteed to alias other pointers. Always use `unsigned char*`, which can always be used to alias other types.

And yes, this is the [Rob Pike pattern](https://commandcenter.blogspot.com/2012/04/byte-order-fallacy.html), and on modern GCC and Clang, this often optimizes straight down to `MOVBE` on Haswell type platforms.